samples/dnn/segmentation.cpp

```diff
@@ -12,81 +12,175 @@ using namespace cv;
+                                         "number of colors (%d != %zu)",
+                                         chns, colors.size()));
+    }
+
+    Mat maxCl = Mat::zeros(rows, cols, CV_8UC1);
+    Mat maxVal(rows, cols, CV_32FC1, score.data);
+    for (int ch = 1; ch < chns; ch++)
+    {
+        for (int row = 0; row < rows; row++)
+        {
+            const float *ptrScore = score.ptr<float>(0, ch, row);
+            uint8_t *ptrMaxCl = maxCl.ptr<uint8_t>(row);
+            float *ptrMaxVal = maxVal.ptr<float>(row);
+            for (int col = 0; col < cols; col++)
+            {
+                if (ptrScore[col] > ptrMaxVal[col])
+                {
+                    ptrMaxVal[col] = ptrScore[col];
+                    ptrMaxCl[col] = (uchar)ch;
+                }
+            }
+        }
+    }
+    segm.create(rows, cols, CV_8UC3);
+    for (int row = 0; row < rows; row++)
+    {
+        const uchar *ptrMaxCl = maxCl.ptr<uchar>(row);
+        Vec3b *ptrSegm = segm.ptr<Vec3b>(row);
+        for (int col = 0; col < cols; col++)
+        {
+            ptrSegm[col] = colors[ptrMaxCl[col]];
+        }
+    }
+}
+
+static void showLegend(FontFace fontFace)
+{
+    static const int kBlockHeight = 30;
+    static Mat legend;
+    if (legend.empty())
+    {
+        const int numClasses = (int)labels.size();
+        if ((int)colors.size() != numClasses)
+        {
+            CV_Error(Error::StsError, format("Number of output labels does not match "
+                                             "number of labels (%zu != %zu)",
+                                             colors.size(), labels.size()));
+        }
+        legend.create(kBlockHeight * numClasses, 200, CV_8UC3);
+        for (int i = 0; i < numClasses; i++)
+        {
+            Mat block = legend.rowRange(i * kBlockHeight, (i + 1) * kBlockHeight);
+            block.setTo(colors[i]);
+            Rect r = getTextSize(Size(), labels[i], Point(), fontFace, 15, 400);
+            r.height += 15; // padding
+            r.width += 10; // padding
+            rectangle(block, r, Scalar::all(255), FILLED);
+            putText(block, labels[i], Point(10, kBlockHeight/2), Scalar(0,0,0), fontFace, 15, 400);
+        }
+        namedWindow("Legend", WINDOW_AUTOSIZE);
+        imshow("Legend", legend);
+    }
+}
 
 int main(int argc, char **argv)
 {
     CommandLineParser parser(argc, argv, keys);
 
     const string modelName = parser.get<String>("@alias");
-    const string zooFile = parser.get<String>("zoo");
+    const string zooFile = findFile(parser.get<String>("zoo"));
 
     keys += genPreprocArguments(modelName, zooFile);
 
     parser = CommandLineParser(argc, argv, keys);
-    parser.about("Use this script to run semantic segmentation deep learning networks using OpenCV.");
-    if (argc == 1 || parser.has("help"))
+    if (!parser.has("@alias") || parser.has("help"))
     {
-        parser.printMessage();
+        cout<<about<<endl;
```

Comment:
It does not print command line options. The command line parser may add custom test to printMessage with ::about(): https://docs.opencv.org/5.x/d0/d2e/classcv_1_1CommandLineParser.html#a2e11e779047efded23f75d9c5c5dd82e